### ISSUE TITLE
User guide adding annotations uses p.renderers.extend instead of p.add_layout

### ISSUE THREAD

**Issue by birdsarah** [bokeh/bokeh#4267]
http://bokeh.pydata.org/en/0.11.1/docs/user_guide/plotting.html#adding-annotations

The source examples:
* https://github.com/bokeh/bokeh/blob/master/sphinx/source/docs/user_guide/source_examples/plotting_span.py
* https://github.com/bokeh/bokeh/blob/master/sphinx/source/docs/user_guide/source_examples/plotting_box_annotation.py

both use 

````
p.renderers.extend([daylight_savings_start, daylight_savings_end])
```

where as https://github.com/bokeh/bokeh/blob/master/sphinx/source/docs/user_guide/source_examples/plotting_label.py 

uses the plot api 

```
p.add_layout(labels)
p.add_layout(citation)
```

I think we should encourage the use of p.add_layout, and update the user guide accordingly.

**Comment by birdsarah** [comment]
Note I updated this issue from `add_annotation` to `add_layout` as I'm removing the `add_annotation` method in a PR I'm working on.

**Comment by bryevdv** [comment]
Should get a deprecation?

**Comment by birdsarah** [comment]
No. See note here: https://github.com/bokeh/bokeh/commit/585d572ad29558909477036939fd18aa7acdf65e

**Comment by birdsarah** [comment]
TL;DR - add_annotation has never been released.

**Comment by birdsarah** [comment]
I was looking at master examples of docs when I spotted the `add_annotation` method and I was looking at 0.11.1 when I spotted the `p.renderers.extend` which originally gave me the impression that it had been released, but I checked and it hasn't.

**Comment by bryevdv** [comment]
Ok agree then. I thought it was in 0.11.1
